### PR TITLE
Interpret command help

### PR DESCRIPTION
**Pull request by dannasessha**
Multi-faceted branch with unit tests, renaming in terminology, expansion of modules/files, and extension of and refactoring of functionality.

### CODE REVIEW

**Comment by dannasessha** [comment]
WIP tag removed, this is ready for review!

File: src/lib.rs [hunk 1]
```diff
@@ -1 +1,252 @@
...
+    let mut command_map = HashMap::new();
+
+    for line in command_help_lines {
+        let (key, value) = define_ident_annotation(line.to_string());
+        command_map.insert(key, value);
+    }
+    command_map
+}
+
+pub fn define_ident_annotation(ident_with_metadata: String) -> (String, String) {
```
Comment (zancas) [inline src/lib.rs:141]: Isn't pairing an `identifier` with a typehint, the act of annotating that identifier?  I like active voice in `fn` names...

So, to me, `annotate_identifier` is quite succinct and active.

What's the benefit of changing `annotate` into `define annotation`?

**Review by zancas — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (zancas) [reply src/lib.rs:141]: Note this comment doesn't block merge.

File: src/lib.rs [hunk 2]
```diff
@@ -1 +1,252 @@
...
+    // find key (String) for hashmap, aka the indentifier
+    let mut ident_temp = ident_with_metadata.trim().split('"').collect::<Vec<&str>>();
+    ident_temp.retain(|&c| c != "");
```
Comment (zancas) [inline src/lib.rs:144]: Ohh...    `retain`..   nice!

File: src/lib.rs (hunk 1 above)
Comment (dannasessha) [reply src/lib.rs:141]: The idea here was that the function was called `annotate_identifier`, but the function defined (parsed out) the identifier. With that name, I would think the identifier would already be defined. I thought about `define_identifier_and_annotation` or something like that, but it seemed too much. Now I think that might be the most clear, even if it's too long.

I can live with whatever terminology we want to settle on.

Comment (zancas) [reply src/lib.rs]: OK...  I understand the naming now that I am looking at `define_annotation`...

**Pull request merged**